packages/plugins/response-cache/src/index.ts

```diff
@@ -22,35 +30,105 @@ function sessionFactoryForEnvelop({ request }: YogaInitialContext) {
   return operationIdByRequest.get(request)
 }
 
+const cacheKeyFactoryForEnvelop: BuildResponseCacheKeyFunction =
+  async function cacheKeyFactoryForEnvelop({ sessionId }) {
+    if (sessionId == null) {
+      throw new Error(
+        '[useResponseCache] This plugin is not configured correctly. Make sure you use this plugin with GraphQL Yoga',
+      )
+    }
+    return sessionId
+  }
+
+const getDocumentStringForEnvelop: GetDocumentStringFunction = (
+  executionArgs,
+) => {
+  const context = executionArgs.contextValue as YogaInitialContext
+  if (context.params?.query == null) {
+    throw new Error(
+      '[useResponseCache] This plugin is not configured correctly. Make sure you use this plugin with GraphQL Yoga',
+    )
+  }
+  return context.params.query as string
+}
+
 export function useResponseCache(options: UseResponseCacheParameter): Plugin {
   const buildResponseCacheKey: BuildResponseCacheKeyFunction =
     options?.buildResponseCacheKey || defaultBuildResponseCacheKey
   const cache = options.cache ?? createInMemoryCache()
   const enabled = options.enabled ?? (() => true)
+  const cachedLastModifiedByRequest = new WeakMap<Request, string>()
+  let logger: YogaLogger
   return {
+    onYogaInit({ yoga }) {
+      logger = yoga.logger
+    },
     onPluginInit({ addPlugin }) {
       addPlugin(
         useEnvelopResponseCache({
           ...options,
           cache,
+          getDocumentString: getDocumentStringForEnvelop,
           session: sessionFactoryForEnvelop,
+          buildResponseCacheKey: cacheKeyFactoryForEnvelop,
+          shouldCacheResult({ result }) {
+            const shouldCached = options.shouldCacheResult
+              ? options.shouldCacheResult({ result })
+              : !result.errors?.length
+            if (shouldCached) {
+              result.extensions ||= {}
+              result.extensions.responseCacheLastModified =
+                new Date().toString()
+            } else {
+              logger.warn('[useResponseCache] Failed to cache due to errors')
```

Comment:
Should this be a warning or more like debug? I feel like errors is something that happens regulary?

Reply:
I saw Envelop logs it as a warning with `console.warn` so I just replicated the same behavior with `logger.warn` here.

Reply:
We can change it later on if it is too annoying and someone complains.